components/NewQuestion.js

```diff
@@ -85,7 +90,7 @@ export default class NewQuestion extends React.Component {
   }
 
   render() {
-    const { queue: { location, fixedLocation } } = this.props
+    const { queue: { location, fixedLocation }, isUserCourseStaff } = this.props
 
     const queueLocation = fixedLocation ? location : this.state.location
 
```

Comment:
One thing (not too big of a deal though), maybe if a user is course staff, change the wording to "Enter your name" to "Enter the student's name".